lib/routes.dart

```diff
@@ -1227,11 +1239,17 @@ extension RouteLinks on RouterState {
   /// If [push] is `true`, then location is pushed to the router location stack.
   void chat(
     ChatId id, {
-    bool push = false,
+    NavigationMode navigationMode = NavigationMode.replaceAll,
     ChatItemId? itemId,
     ChatDirectLinkSlug? link,
   }) {
-    (push ? this.push : go)('${Routes.chats}/$id');
+    if (navigationMode == NavigationMode.replace) {
+      routes.removeLast();
+    }
+
+    (navigationMode == NavigationMode.replaceAll ? go : push)(
+      '${Routes.chats}/$id',
+    );
```

Comment:
Perhaps a `switch (mode) { ... }` would be cleaner for this method? It's a bit complicated to read it currently.